sisl/geom/surfaces.py

```diff
@@ -0,0 +1,587 @@
+                         atoms=atoms, sc=sc)
+            g = g.tile(info.nlayers, 2)
+
+            # slide ABC layers relative to each other
+            B = 2 * (info.offset + 1) % 6
+            C = 2 * (info.offset + 2) % 6
+            vec = (3 * sc.cell[0] + sc.cell[1]) / 6
+            g.xyz[B::6] += vec
+            g.xyz[B+1::6] += vec
+            g.xyz[C::6] += 2 * vec
+            g.xyz[C+1::6] += 2 * vec
+
+        else:
+            sc = SuperCell(np.array([[0.5, 0, 0],
+                                     [0.125, 0.375, 0],
+                                     [0, 0, 1 / 3]]) ** 0.5 * alat)
+            g = Geometry([0, 0, 0], atoms=atoms, sc=sc)
+            g = g.tile(info.nlayers, 2)
+
+            # slide ABC layers relative to each other
+            B = (info.offset + 1) % 3
+            C = (info.offset + 2) % 3
+            vec = (sc.cell[0] + sc.cell[1]) / 3
+            g.xyz[B::3] += vec
+            g.xyz[C::3] += 2 * vec
+
+    else:
+        raise NotImplementedError(f"fcc_slab: miller={miller} is not implemented")
+
+    g = _finish_slab(g, vacuum)
+    return g
+
+
+def bcc_slab(alat, atoms, miller, layers=None, vacuum=20., *, orthogonal=False, start=None, end=None):
+    r""" Construction of a surface slab from a body-centered cubic (BCC) crystal
+
+    The slab layers are stacked along the :math:`z`-axis. The default stacking is the first
+    layer as an A-layer, defined as the plane containing an atom at :math:`(x,y)=(0,0)`.
+
+    Parameters
+    ----------
+    alat : float
+        lattice constant of the fcc crystal
+    atoms : Atom
+        the atom that the crystal consists of
+    miller : int or str or 3-array
+        Miller indices of the surface facet
+    layers : int or str, optional
+        Number of layers in the slab or explicit layer specification.
+        An empty character `' '` will be denoted as a vacuum slot, see examples.
+        Currently the layers cannot have stacking faults.
+    vacuum : float or array_like, optional
+        distance added to the third lattice vector to separate
+        the slab from its periodic images. If this is None, the slab will be a fully
+        periodic geometry but with the slab layers. Useful for appending geometries together.
+        If an array layers should be a str, it should be no longer than the number of spaces
+        in `layers`. If shorter the last item will be repeated (like `zip_longest`).
+    orthogonal : bool, optional
+        if True returns an orthogonal lattice
+    start : int or string, optional
+        sets the first layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+    end : int or string, optional
+        sets the last layer in the slab. Only one of `start` or `end` must be specified.
+        If set together with `layers` being a str, then they *must* be conforming.
+
+    Examples
```

Comment:
Maybe we could shorten this and refer to the `fcc_slab` examples (to avoid duplication)?